pkg/reconciler/accessor/networking/certificate.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/logging"
+	"knative.dev/serving/pkg/apis/networking/v1alpha1"
+	clientset "knative.dev/serving/pkg/client/clientset/versioned"
+	listers "knative.dev/serving/pkg/client/listers/networking/v1alpha1"
+	kaccessor "knative.dev/serving/pkg/reconciler/accessor"
+)
+
+// CertificateAccessor is an interface for accessing Knative Certificate.
+type CertificateAccessor interface {
+	GetServingClient() clientset.Interface
+	GetCertificateLister() listers.CertificateLister
+}
+
+// ReconcileCertificate reconciles Certificate to the desired status.
+func ReconcileCertificate(ctx context.Context, owner kmeta.Accessor, desired *v1alpha1.Certificate,
+	certAccessor CertificateAccessor) (*v1alpha1.Certificate, error) {
```

Comment:
I don't see the need for an Accessor interface

Why not just pass an instance of `clientset.Interface` and `listers.CertificateLister` to `ReconcileCertificate` ?

Reply:
The `CertificateAccessor` is a wrapper that contains the functionalities needed by `ReconcileCertificate`.
In future, if we need to adjust the logic of `ReconcileCertificate` and need to pass more functionalities into `ReconcileCertificate`, we can simply modify `CertificateAccessor` without changing the inputs of `ReconcileCertificate`.